tweak: rewrite kpi component, canvas cleanup

briangregoryholmes opened the pull request:
<img width="1247" alt="Screenshot 2025-02-19 at 8 56 41 PM" src="https://github.com/user-attachments/assets/1570a43b-18e0-41a2-a85c-618bf2bea541" />
<img width="1247" alt="Screenshot 2025-02-19 at 8 56 59 PM" src="https://github.com/user-attachments/assets/527b1619-4c4a-4b6e-802b-63e1b230c8e6" />


**Checklist:**
- [ ] Covered by tests
- [x] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [ ] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [x] Checked if the docs need to be updated
- [x] Intend to cherry-pick into the release branch
- [x] I'm proud of this work!

djbarnwal commented:
Great work! Looks neat and exciting to see the new changes to KPI. Some initial thoughts - 

UXQA
1. Charts size is broken (blocker)
2. We need an empty state for the KPI
3. When local time filters are applied to KPI without a local comparison range, do not inherit global comparison (blocker)

https://jam.dev/c/417355de-240e-4002-902b-95822763247d


<img width="217" alt="image" src="https://github.com/user-attachments/assets/cadd400d-c805-4897-90b2-2164af45f698" />

This feels a bit too dense. I wonder if we can add some padding on both sides of the `/` separator.

<img width="217" alt="image" src="https://github.com/user-attachments/assets/e9fa879c-fc9a-415a-a053-e16bc3123007" />

Hide blue dot when null

djbarnwal reviewed (changes requested):
Blockers

- Expanding chart container
- Unable to set local comparison range for KPI grid
- Do not inherit global range when local time range is set

briangregoryholmes commented:
- Hid points when null
- Added padding between total hovered value
- Resolved chart container issue, though needs some follow up tweaks
- Updated the conditional on when to show comparisons

`Unable to set local comparison range for KPI grid`

This only happens on first load and seems to be an issue in `canvas-entity.ts`. Can you take a look at this one and hold off on blocking this PR?

djbarnwal reviewed (commented):
UXQA:

The main line dot should be above comparison dot - 
![image](https://github.com/user-attachments/assets/4fd8c429-c1eb-4af7-80d5-c4519fadf030)

KPI layout is broken when sparkline is placed on the right

![image](https://github.com/user-attachments/assets/58369a40-ad84-45c0-9a1e-dfab39040f14)

If primary point is null and comparison point has a value, `no change` is incorrectly displayed

![image](https://github.com/user-attachments/assets/04d1209f-2f6d-4646-bd2d-44d0569004bd)

Adding a title breaks layout and increase row height 
![image](https://github.com/user-attachments/assets/c4ec81cc-c947-42be-a27d-89cf8e8d95e1)
